src/Agent.Worker/TaskManager.cs

```diff
@@ -83,6 +93,59 @@ into taskGrouping
                 }
             }
         }
+        public async Task RemoveNodeRunner(IExecutionContext executionContext)
```

Comment:
I'm not sure we need to remove node6.  We should not on managed agents, and it might not be necessary to remove on selfhosted